AK/HashTable.h

```diff
@@ -501,10 +500,10 @@ class HashTable {
         for (;;) {
             auto& bucket = m_buckets[hash % m_capacity];
 
-            if (bucket.used && predicate(*bucket.slot()))
+            if (bucket.state == BucketState::Used && predicate(*bucket.slot()))
                 return &bucket;
 
-            if (!bucket.used && !bucket.deleted)
+            if (bucket.state != BucketState::Used && bucket.state != BucketState::Deleted)
```

Comment:
Since this is done a couple of times we might be able to pick some specific values for the enums such that this check can be quicker? Definitely not needed in this PR just a (fun?) micro optimization.

Reply:
I'll put a FIXME up with the BucketState definition.